### DISCUSSION TITLE
POpenFOAMReader: reading particles' data?

### DISCUSSION THREAD

**Discussion by TessellateDataScience** [pyvista/pyvista#6122] [Q&A] [answered]
### Describe the feature you would like to be added.

I'm aware that PyVista allows reading of OpenFOAM fluid-related data. Does PyVista currently have the ability to read (and plot) particle-related data that FOAM spit's out after doing it's particle tracking'?

**Comment by tkoyama010** [comment]
I don't think pv.`POpenFOAMReader` has it; if VTK has the corresponding read function, we can consider implementing it.

**Reply by MatthewFlamm** [reply, marked as answer]
vtkOpenFOAMReader, which is a parent class of vtkPOpenFOAMReader, states it has "lagrangian field support".  See [vtk docs](https://vtk.org/doc/nightly/html/classvtkOpenFOAMReader.html#details).

Have you tried it?

**Reply by TessellateDataScience** [reply]
Thanks folks! I'm happy to help if I can with implementing this (as I asked for it).

**Reply by MatthewFlamm** [reply]
I would start by trying to read your data with the existing code and reporting back.  It is quite possible that it already works.

**Reply by TessellateDataScience** [reply]
Sorry Matthew, busy creating data first. Tried using vtkOpenFOAMReader, but can't read the particle data (I can read fluid data). It's not due to particle's directory structure. Data is output from OpenFOAM-dev: https://github.com/OpenFOAM/OpenFOAM-dev/commit/fb4d7abf4217673f9b03a3d0d70c45524c79bd12

**Reply by MatthewFlamm** [reply]
Not sure whether you are saying that the directory structure changed or the data has changed.  In either case, if the vtkOpenFOAMReader or vtkPOpenFOAMReader needs to be updated, this would have to be done in vtk package.

**Comment by TessellateDataScience** [comment]
Also, I don't need use of decomposed functionality in pyvista (i.e. I can reconstruct my case fully beforehand then import that as single entity** without using `vtkPOpenFOAMReader`). Use of large meshes is overkill for us, as we are a very underfunded non-profit organisation with limited time and capacity thus we don't do hi-fi computations such as DNS.

** May not be correct software-dev term, sorry.
